fe1-web/model/network/method/message/data/election/SetupElection.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Hash, Timestamp, Lao, EventTags,
+} from 'model/objects';
+import { OpenedLaoStore } from 'store';
+import { ProtocolError } from 'model/network/ProtocolError';
+import { validateDataObject } from 'model/network/validation';
+import { Question } from 'model/objects/Election';
+import { ActionType, MessageData, ObjectType } from '../MessageData';
+import { checkTimestampStaleness } from '../Checker';
+
+export class SetupElection implements MessageData {
+  public readonly object: ObjectType = ObjectType.ELECTION;
+
+  public readonly action: ActionType = ActionType.SETUP;
+
+  public readonly lao: string;
+
+  public readonly id: Hash;
+
+  public readonly name: string;
+
+  public readonly version: string;
+
+  public readonly created_at: Timestamp;
+
+  public readonly start_time: Timestamp;
+
+  public readonly end_time: Timestamp;
+
+  public readonly questions: Question[];
+
+  constructor(msg: Partial<SetupElection>) {
+    if (!msg.id) throw new ProtocolError('Undefined \'id\' parameter encountered during \'SetupElection\'');
+
+    if (!msg.lao) throw new ProtocolError('Undefined \'lao\' parameter encountered during \'SetupElection\'');
+    this.lao = msg.lao;
+
+    if (!msg.name) throw new ProtocolError('Undefined \'name\' parameter encountered during \'SetupElection\'');
+    this.name = msg.name;
+
+    if (!msg.version) throw new ProtocolError('Undefined \'version\' parameter encountered during \'SetupElection\'');
+    this.version = msg.version;
+
+    if (!msg.created_at) throw new ProtocolError('Undefined \'created_at\' parameter encountered during \'SetupElection\'');
+    checkTimestampStaleness(msg.created_at);
+    this.created_at = msg.created_at;
+    if (!msg.start_time) throw new ProtocolError('Undefined \'start_time\' parameter encountered during \'SetupElection\'');
+    checkTimestampStaleness(msg.start_time);
+    if (!msg.end_time) throw new ProtocolError('Undefined \'end_time\' parameter encountered during \'SetupElection\'');
+    checkTimestampStaleness(msg.end_time);
+    if (msg.end_time < msg.created_at) throw new ProtocolError('Invalid timestamp encountered: \'end\' parameter smaller than \'creation\'');
+    if (msg.end_time < msg.start_time) throw new ProtocolError('Invalid timestamp encountered: \'end\' parameter smaller than \'start\'');
+    this.end_time = msg.end_time;
+    if (msg.created_at > msg.start_time) {
+      // This avoids errors when the start-time is set before creation time (Which is invalid)
+      // that can happen unwillingly when the organizer takes time creating an election
+      this.start_time = msg.created_at;
+    } else {
+      this.start_time = msg.start_time;
+    }
```

Comment:
This might be called when receiving a message from the network though, thus "corrupting" its original content.

Is there a better place to address this issue? (closer to the source of the problem, perhaps)

Reply:
Good point! I didn't think about that being called when getting a message from the network.
I will do it at the source in the creation.

Reply:
Now its implemented as a ternary operator in the MessageApi.ts 
```
start_time: ((start > time) ? start : time),
```